src/main/java/randoop/types/Type.java

```diff
@@ -72,6 +73,82 @@ public static Type forName(@ClassGetName String typeName) throws ClassNotFoundEx
     return Type.forClass(c);
   }
 
+  /**
+   * Returns the Type for a fully qualified name (that may or may not be a multi-dimensional array).
+   *
+   * @param fullyQualifiedName the fully qualified name of a type. Array names such as {@code int[]}
+   *     or {@code java.lang.String[][]} are also fully qualified names. However, array definitions
+   *     with sizes, such as {@code int[3][]}, are not fully qualified names.
+   * @return the type object for the type with the name, null if none is found
+   * @throws ClassNotFoundException if name is not a recognized type
+   */
+  public static Type getTypeforFullyQualifiedName(@ClassGetName String fullyQualifiedName)
+      throws ClassNotFoundException {
+    Class<?> className = forFullyQualifiedName(fullyQualifiedName);
+    return className.isArray() ? ArrayType.forClass(className) : Type.forClass(className);
+  }
+
+  /**
+   * Returns the Class for a fully qualified name (that may or may not be a multi-dimensional
+   * array).
+   *
+   * @param fullyQualifiedName the fully qualified name of a type. Array names such as {@code int[]}
+   *     or {@code java.lang.String[][]} are also fully qualified names. However, array definitions
+   *     with sizes, such as {@code int[3][]}, are not fully qualified names.
+   * @return the type object for the type with the name, null if none is found
+   * @throws ClassNotFoundException if name is not a recognized type
+   */
+  public static Class<?> forFullyQualifiedName(@ClassGetName String fullyQualifiedName)
+      throws ClassNotFoundException {
+    String[] fullyQualifiedArrayParsedName = fullyQualifiedName.split("\\[");
+    int arrayDimension = fullyQualifiedArrayParsedName.length - 1;
+    Class<?> fullyQualifiedBaseType =
+        forFullyQualifiedNameNonArray(fullyQualifiedArrayParsedName[0]);
+
+    if (arrayDimension > 0) {
+      // Make each dimension size zero, since it is ignored by getClass().
+      int[] dimensions = new int[arrayDimension];
+      return Array.newInstance(fullyQualifiedBaseType, dimensions).getClass();
+    } else {
+      return fullyQualifiedBaseType;
+    }
+  }
+
+  /**
+   * Returns the Class for a fully qualified name. Does not support arrays.
+   *
+   * @param fullyQualifiedName the fully qualified name of a non-array type
+   * @return the type object for the type with the name, null if none is found
+   * @throws ClassNotFoundException if name is not a recognized type
```

Comment:
The above two lines seem to be in conflict.  Does it return null or throw an exception?  Or do both in different situations (which would be a bit confusing)?